From 6213095d7ec6c153515d22b3b441d0aeed67c87a Mon Sep 17 00:00:00 2001
From: Maksim Kuznetsov <mk@4xxi.com>
Date: Fri, 17 Feb 2017 07:46:51 +0300
Subject: [PATCH] Fix sharing folders having similar names

---
 recipe/deploy/shared.php   |  2 +-

diff --git a/recipe/deploy/shared.php b/recipe/deploy/shared.php
index c194dd1..16b35f7 100644
--- a/recipe/deploy/shared.php
+++ b/recipe/deploy/shared.php
@@ -16,7 +16,7 @@
     // Validate shared_dir, find duplicates
     foreach (get('shared_dirs') as $a) {
         foreach (get('shared_dirs') as $b) {
-            if ($a !== $b && strpos($a, $b) === 0) {
+            if ($a !== $b && strpos(rtrim($a, '/') . '/', rtrim($b, '/') . '/') === 0) {
                 throw new ConfigurationException("Can not share same dirs `$a` and `$b`.");
             }
         }
